tests/SpecTests/PrimaryStepDownSpecTest.php

```diff
@@ -0,0 +1,278 @@
+<?php
+
+namespace MongoDB\Tests\SpecTests;
+
+use IteratorIterator;
+use MongoDB\Client;
+use MongoDB\Collection;
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\BulkWriteException;
+use MongoDB\Driver\Exception\ConnectionTimeoutException;
+use MongoDB\Driver\Exception\Exception as DriverException;
+use MongoDB\Driver\ReadPreference;
+use MongoDB\Driver\WriteConcern;
+use MongoDB\Operation\BulkWrite;
+use MongoDB\Tests\CommandObserver;
+use Symfony\Bridge\PhpUnit\SetUpTearDownTrait;
+use UnexpectedValueException;
+
+/**
+ * @see https://github.com/mongodb/specifications/tree/master/source/connections-survive-step-down/tests
+ */
+class PrimaryStepDownSpecTest extends FunctionalTestCase
+{
+    use SetUpTearDownTrait;
+
+    /** @var Client */
+    private $client;
+
+    /** @var Collection */
+    private $collection;
+
+    private function doSetUp()
```

Comment:
I was going to suggest dropping the collection under test on tearDown, since the tests write to it; however, I realized that we don't consistently do that across the test suite. For example: [`MongoDB\Tests\Collection\FunctionalTestCase::tearDown()`[(https://github.com/mongodb/mongo-php-library/blob/1.4.3/tests/Collection/FunctionalTestCase.php#L25) drops collections after a successful test and leaves behind data after a failure in case we need to inspect it.

I don't think this is worth doing here, but I did want to share the thought. Long term, I'd be more interested in auditing/profiling the test suite to see where we can improve execution time by _reducing_ things like drops. I know database drops can be fairly expensive, so those would be worth looking at to see if they serve a useful purpose.

Reply:
I'll leave it for now. As you mentioned, leaving data behind to check out after test failures can make sense, and the spec tests drop collections during setup anyways. At a previous job I've played around with only dropping collections that were written to during tests (using APM to figure out which collections were written to), but the performance gain from that was negligible IIRC.